src/hotspot/share/cds/filemap.cpp

```diff
@@ -2409,6 +2452,11 @@ bool FileMapHeader::validate() {
   // header data
   const char* prop = Arguments::get_property("java.system.class.loader");
   if (prop != nullptr) {
+    if (has_aot_linked_classes()) {
```

Comment:
Should this check be part of `FileMapInfo::validate_aot_class_linking`?

Reply:
I put the check here so that the "Archived non-system classes are disabled because ...." message will not be printed a few lines below. Otherwise the user will see two different error messages that say different things.